src/control/server/server.go

```diff
@@ -147,12 +147,29 @@ func Start(log *logging.LeveledLogger, cfg *Configuration) error {
 	}
 
 	// Create new grpc server, register services and start serving.
+	var opts []grpc.ServerOption
 	tcOpt, err := security.ServerOptionForTransportConfig(cfg.TransportConfig)
 	if err != nil {
 		return err
 	}
+	opts = append(opts, tcOpt)
 
-	grpcServer := grpc.NewServer(tcOpt)
+	uintOpt, err := unaryInterceptorForTransportConfig(cfg.TransportConfig)
+	if err != nil {
+		return err
+	}
+	if uintOpt != nil {
+		opts = append(opts, uintOpt)
+	}
+	sintOpt, err := streamInterceptorForTransportConfig(cfg.TransportConfig)
```

Comment:
Just a question, why are these interceptor options in the server module, but the ServerOptionForTransportConfig is coming from the security module? Should they all be in the same module?

Reply:
The options for transport config have both client and server components however there is only server side interceptors. Also the interceptors arent security specific. We can add other things like logging in there as well. The only reason I pass in the transport config is because that is where we know if we're in insecure mode or not. I guess I could do the check here but it is more in line with what we did for the server/client options.

Reply:
I think it's OK as-is. Thanks for the explanation.